### DISCUSSION TITLE
Augmentation Method 0

### DISCUSSION THREAD

**Discussion by papygreg** [JSBSim-Team/jsbsim#743] [Support requests]
Hi folks !
I have tried to tie the augmentation method 0 to the FCS using the following lines :

```xml
<!-- engine 1 augmentation -->
			<switch name="names/throttle/augmentation">
				<default value="0"/>
					<test logic="AND" value="1">
					  fcs/throttle-pos-norm ge 75
					  propulsion/engine/n2 ge 84
					  /engines/engine/running == 1
				  </test>
				<output>/engines/engine/augmentation</output>
			</switch>
			<!-- engine 2 augmentation -->
			<switch name="names/throttle1/augmentation">
				<default value="0"/>
					<test logic="AND" value="1">
					  fcs/throttle-pos-norm ge 75
					  propulsion/engine[1]/n2 ge 84
					  /engines/engine[1]/running == 1
				  </test>
				<output>/engines/engine[1]/augmentation</output>
			</switch>
````

Unfortunnatly nothing happens. I think I use a wrong path but i am unable to find the correct one.
Is anyone familiar with this issue ?
many thanks !

**Comment by dpculp** [comment]
Throttle-pos-norm is a value in the normalized range, 0.0 to 1.0, so it
will never get to 75.

**Reply by papygreg** [reply]
Yes, i have tried also with values between 0.0 and 1.0 without any results

**Reply by bomber64** [reply]
Are the output properties correct ?

**Reply by papygreg** [reply]
Yes, we track the throttle-pos-norm value as well as the augmentation bool value. The throttle-pos-norm evolves correctly and is consistent with the throttle position we send to jsbsim. The augmentation bool stays at 0 at all times even if the conditions in the switch are reached

**Reply by bomber64** [reply]
I've not done augmentation before but I  must admit I'd have expected JSBsim to have put the augmentation property in propulsion/engine[0] but of you say it's in engines/engine[0]..... Then who am I to say otherwise.

**Reply by papygreg** [reply]
The path is given in the FGTurbine source

**Comment by seanmcleod** [comment]
You haven't specified which engine you're using or shown the relevant snippet from your engine file around the augmentation method.

https://github.com/JSBSim-Team/jsbsim/blob/f13d015dfad49be306e4f313b5e7d1406205ea64/src/models/propulsion/FGTurbine.h#L133-L139

So depending on the augmentation method specified the FGTurbine code could be updating it on each time step.

https://github.com/JSBSim-Team/jsbsim/blob/f13d015dfad49be306e4f313b5e7d1406205ea64/src/models/propulsion/FGTurbine.cpp#L231-L252

So the first thing to confirm is what augmentation method you have specified in your engine file. You have mentioned in your original post that you're trying to use augmentation method 0, but seeing is believing 😉

**Reply by papygreg** [reply]
There you go

```xml
-- File: F119-PW-100.xml Author: Aero-Matic v 0.8 Inputs: name: F119-PW-100 type: turbine thrust: 37000.8 lb augmented? yes injected? no -->

<turbine_engine name="F119-PW-100">

<milthrust> 26950.0 </milthrust>

<maxthrust> 37000.0 </maxthrust>

<bypassratio> 0.2 </bypassratio>

<tsfc> 0.8 </tsfc>

<atsfc> 1.95 </atsfc>

<idlen1> 40.0 </idlen1>

<idlen2> 53.0 </idlen2>

<maxn1> 100.0 </maxn1>

<maxn2> 100.0 </maxn2>

<augmented> 1 </augmented>

<augmethod> 0 </augmethod>

<injected> 0 </injected>

```
It is a turbine engine set with the good parameters. I have put the switch function in the throttle Control part

**Reply by papygreg** [reply]
In fact, I just realised augmentation methods 0 & 1 are all or nothing. So I'm digging into the augmentation method 2. Then, the question is how to expand the throttle-pos range in the fcs. Digging through the code get me to realize only value between 0 and 1 are allowed for the throttle-pos

**Comment by dpculp** [comment]
The T-38 model in the aircraft repository is an example of how to use
augmentation method 2.  In this case the throttle command (0.0-1.0) is
multiplied by 2 using a pure_gain FCS element.  So the value is now
mapped to the throttle position in the range 0.0-2.0.  The turbine
model will now treat the range 0.0-1.0 as unaugmented, and the range
>1.0-2.0 as augmented.

You can change the multiplier value in step one above to alter the
proportion of your physical throttle range that will be allotted for
afterburner use.

**Reply by papygreg** [reply]
Thanks, I will check that tomorrow!
